src/main/java/teammates/ui/webapi/CreateFeedbackResponseCommentAction.java

```diff
@@ -54,6 +54,7 @@ void checkSpecificAccessControl() throws UnauthorizedAccessException {
             if (studentAttributes == null) {
                 throw new EntityNotFoundException("Student does not exist.");
             }
+            session = session.sanitizeForStudent(studentAttributes.getEmail());
```

Comment:
At the time of review, I was thinking this is done out of security concern regardless the API, such that even if it doesn't matter right now, any future changes to this API will not result in exposing data from the session that a student should not view.

Reply:
I believe this updates the session with the individual deadline of the student, which is used in line 60 when checking if the session is still open.

Reply:
It is necessary to ensure that the session uses the correct deadline for the users, like whether it is extended or not, when checking if the session is open or closed and whatnot by sanitizing it like this. I searched for usages of `verifySessionOpenExceptForModeration(session)` and I added this sanitization for all the usages, including this. However, I am not sure if it was really necessary to do it for all of them. If you think it is not necessary for this case, it can be removed.

Reply:
I am not well-versed enough with the logic, but judging from what you described, this method should not be named "sanitize". Sanitize is for updating values such that they are secure (think XSS/SQL injection). Even obscuring/hiding information should not be named "sanitize", much less conditionally updating the values.

Reply:
At first, I thought of using `sanitize` and suggested, so the name is as such. As you pointed out, it does sound inappropriate for its functionality.

The gist is to alter the `endTime` for different user and hide away the deadlines from other users if the user should not view them. What if we call it `maskForStudent` or `prepareForStudent`?

Reply:
I don't even think this attribute should be concerned with any of them. Ensuring that the HTTP response is correct and unnecessary information obfuscated is the duty of the `*Data` object.

And especially if you see that this is access control method: nothing from this method goes out to the HTTP response.

If the intention is to ensure the correct deadline is reflected in the attributes, you should probably dedicate the method for such purpose.

Reply:
> If the intention is to ensure the correct deadline is reflected in the attributes, you should probably dedicate the method for such purpose.

Yes, actually the sanitization parts have already been moved to the `*Data` object. This method is only concerned with ensuring the correct deadline is reflected in attributes. @jayasting98, let's think of a suitable name and change the name in the next PR!